src/WeatherSensor.cpp

```diff
@@ -806,25 +806,20 @@ DecodeStatus WeatherSensor::decodeBresser7In1Payload(uint8_t *msg, uint8_t msgSi
       log_e("DECODE_FAIL_SANITY !!!");
   }
   // data whitening
-  for (unsigned i = 0; i < msgSize; ++i) {
-      msg[i] ^= 0xaa;
+  uint8_t msgw[msgSize];
```

Comment:
I wonder why this passed the CI... msgSize is a parameter to the function, hence it is not known at compile time!
In my opinion this should not work. Maybe I'm missing something here...

The origin is here: https://github.com/matthias-bs/BresserWeatherSensorReceiver/blob/9a75f32f63941ed5722b5d9c91d003eea68c6a7a/src/WeatherSensor.cpp#L198

It would not harm to put it into a #define instead of using the constant number 27  again.